client/state-db/src/lib.rs

```diff
@@ -97,12 +97,19 @@ impl<
 {
 }
 
-/// Backend database trait. Read-only.
+/// Backend database trait.
 pub trait MetaDb {
 	type Error: fmt::Debug;
 
 	/// Get meta value, such as the journal.
 	fn get_meta(&self, key: &[u8]) -> Result<Option<DBValue>, Self::Error>;
+
+	/// Set meta value
```

Comment:
```suggestion
	/// Write meta key-value to the database. This is only used when opening the database to initialise the metadata.
```

Reply:
Such "pinky promise" in the doc-comment makes me want to return two separate traits (for `read` and `write`) back.

Reply:
In a more formal language: it isn't a terribly good practice for this `MetaDb` trait to be aware of why it is used (i.e. the difference between opening/initialisation and further use).

Reply:
`MetaDb` an interface required by `StateDb` specifically. It is not some generic trait to be used by any other system. So I think `StateDb` may declare how exactly it is used to plug in the database.

I now think a better option would be for `open` to return an `Option<CommitSet>` that would be written by the caller instead. That would be consistent with the general `StateDb` design, where it does not actively update the database, but only produces change sets.

Reply:
Oh! I like that! That indeed would be "in the spirit" of the rest of the crate :)